core/com.dexels.navajo.function/src/com/dexels/navajo/functions/Max.java

```diff
@@ -86,20 +93,27 @@ private Object max(Object a, Object b)
         }
         else
         {
-            if ((b instanceof Date))
+            if ( b != null && ( b instanceof Date || b instanceof ClockTime ) )
             {
-                throw new TMLExpressionException("Cannot compare a date with a non-date");
+                if( a == null )
+                {
+                    return b;
+                }
+                else
+                {
+                    throw new TMLExpressionException( "Cannot compare a date/clocktime with a non-date/clocktime" );
+                }
             }
             else
             {
-                double value1 = Utils.getDoubleValue(a);
-                double value2 = Utils.getDoubleValue(b);
-                double max = (value1 > value2) ? value1 : value2;
+                double value1 = Utils.getDoubleValue( a );
+                double value2 = Utils.getDoubleValue( b );
+                double max = ( value1 > value2 ) ? value1 : value2;
 
-                if (a instanceof Double || b instanceof Double)
-                    return Double.valueOf(max);
+                if ( a instanceof Double || b instanceof Double )
+                    return Double.valueOf( max );
                 else
-                    return Integer.valueOf((int) max);
+                    return Integer.valueOf( (int) max );
```

Comment:
Please don't mix features with code formatting

Reply:
I don't agree. When new code is formatted a different way than existing code, the rest should be adjusted to match. Maybe it should've been in a separate commit so you can ignore the formatting commit, but every formatting change has a chance to introduce errors so should not be ignored at a review anyways.

Whether it's "correct" to format it my way is another discussion that we can have ;)